cmd/juju/application/removeunit.go

```diff
@@ -144,38 +193,57 @@ func (c *removeUnitCommand) unitsHaveStorage(unitNames []string) (bool, error) {
 // Run connects to the environment specified on the command line and destroys
 // units therein.
 func (c *removeUnitCommand) Run(ctx *cmd.Context) error {
+	logger.Criticalf("here we are cc")
 	client, apiVersion, err := c.getAPI()
 	if err != nil {
 		return err
 	}
 	defer client.Close()
 
+	logger.Criticalf("here we are 1")
 	if apiVersion < 4 {
 		return c.removeUnitsDeprecated(ctx, client)
 	}
+
+	logger.Criticalf("here we are 2")
+	if err := c.validateRemovalByModelType(); err != nil {
+		return errors.Trace(err)
+	}
+
+	logger.Criticalf("here we are 3")
+	modelType, err := c.ModelType()
```

Comment:
I think validateRemovalByType() could take model type as an arg since we get it inside the method and also again here.